File: netlify/functions/songs-add.js
// netlify/functions/songs-add.js
const { loadEnv } = require('./utils/load-env'); // Load environment variables from appropriate .env file
loadEnv();

const { createClient } = require('@supabase/supabase-js');
const { headers } = require('./utils/cors-headers');

exports.handler = async function(event, context) {
  // Only allow POST requests
  if (event.httpMethod !== "POST") {
    return {
      statusCode: 405,
      headers: {
        ...headers,
        "Allow": "POST"
      },
      body: JSON.stringify({ error: "Method not allowed" })
    };
  }
  
  try {
    // Get Supabase credentials from environment variables
    const supabaseUrl = process.env.SUPABASE_URL;
    const supabaseKey = process.env.SUPABASE_ANON_KEY;
    
    if (!supabaseUrl || !supabaseKey) {
      console.error('Missing Supabase credentials');
      return {
        statusCode: 500,
        headers,
        body: JSON.stringify({ 
          error: 'Server configuration error - missing Supabase credentials' 
        })
      };
    }

    // Initialize Supabase client
    const supabase = createClient(supabaseUrl, supabaseKey);
    
    // Parse request body
    const songData = JSON.parse(event.body);
    
    // Basic validation
    if (!songData || !songData.queue_id || !songData.song_id || !songData.user_id) {
      return {
        statusCode: 400,
        headers,
        body: JSON.stringify({ 
          error: "Invalid song data", 
          detail: "Missing required fields (queue_id, song_id, or user_id)" 
        })
      };
    }
    
    console.log('Adding song to queue:', songData);
    
    // Get current timestamp
    const timestamp = new Date().toISOString();
    
    // Create a song record
    const { data, error } = await supabase
      .from('queue_songs')
      .insert([
        { 
          queue_id: songData.queue_id,
          song_id: songData.song_id,
          added_by: songData.user_id,
          created_at: timestamp,
          status: 'pending'
        }
      ])
      .select();
    
    if (error) {
      console.error('Error adding song to queue:', error);
      return {
        statusCode: 500,
        headers,
        body: JSON.stringify({ 
          error: "Failed to add song to queue", 
          detail: error.message 
        })
      };
    }
    
    // Return success response
    return {
      statusCode: 201,
      headers,
      body: JSON.stringify({ 
        success: true,
        message: "Song added to queue successfully",
        song: data[0]
      })
    };
  } catch (error) {
    console.error('Unexpected error in songs-add function:', error);
    return {
      statusCode: 500,
      headers,
      body: JSON.stringify({ 
        error: "Failed to add song",
        message: error.message 
      })
    };
  }
};

File: netlify/functions/songs-add/songs-add.py
# netlify/functions/songs-add/songs-add.py
import sys
import os
import json
import uuid

# Add the deps directory to the Python path
current_dir = os.path.dirname(os.path.abspath(__file__))
sys.path.append(os.path.join(current_dir, "deps"))

# Now import your dependencies
import requests
from dotenv import load_dotenv
from supabase import create_client

# Load environment variables
load_dotenv()

# Initialize Supabase client
SUPABASE_URL = os.environ.get("SUPABASE_URL")
SUPABASE_KEY = os.environ.get("SUPABASE_ANON_KEY")

supabase = None
if SUPABASE_URL and SUPABASE_KEY:
    try:
        supabase = create_client(SUPABASE_URL, SUPABASE_KEY)
    except Exception as e:
        print(f"Error initializing Supabase client: {e}")

def handler(event, context):
    """
    Netlify Function handler for adding songs to a queue
    """
    try:
        if not supabase:
            return {
                'statusCode': 500,
                'body': json.dumps({'error': 'Supabase client not initialized'})
            }
        
        # Parse the request body
        body = json.loads(event['body']) if event.get('body') else {}
        queue_id = body.get('queueId', '')
        song_id = body.get('songId', '')
        user_id = body.get('userId', '')
        
        if not queue_id or not song_id:
            return {
                'statusCode': 400,
                'body': json.dumps({'error': 'Missing required parameters'})
            }
        
        # Validate queue_id is a valid UUID
        try:
            uuid.UUID(queue_id)
        except ValueError:
            return {
                'statusCode': 400,
                'body': json.dumps({'error': 'Invalid queue ID format'})
            }
        
        # Here you would implement the song addition logic
        # For example, insert into Supabase
        
        # For now, return a mock success response
        return {
            'statusCode': 200,
            'body': json.dumps({
                'success': True,
                'data': {
                    'id': str(uuid.uuid4()),
                    'queueId': queue_id,
                    'songId': song_id,
                    'createdAt': '2025-03-03T22:00:00Z'
                }
            })
        }
    except Exception as e:
        return {
            'statusCode': 500,
            'body': json.dumps({'error': str(e)})
        }
